odoo/models.py

```diff
@@ -3810,15 +3803,22 @@ def create(self, vals_list):
             # add missing defaults
             vals = self._add_missing_default_values(vals)
 
+            # set magic fields
+            for name in bad_names:
+                vals.pop(name, None)
+            if self._log_access:
+                vals.setdefault('create_uid', self.env.uid)
+                vals.setdefault('create_date', self.env.cr.now())
+                vals.setdefault('write_uid', self.env.uid)
+                vals.setdefault('write_date', self.env.cr.now())
```

Comment:
That's a change in behaviour: previously, create_uid and create_date could not be overwritten (though write_uid and write_date could, for some weird reason).

Reply:
Look at the two lines before that code snippet: they will discard from `vals` the magic fields that cannot be assigned.  Overall there should be no change in behavior.

Reply:
Then what's the point of using `setdefault`? That sends the wrong message to the reader.

Reply:
ah I see from the comment which was out-of-diff, nm